### PR TITLE
[language][move] Added typing tests [part 5]

### PR DESCRIPTION
**Pull request by tnowacki**
## Motivation

- Added tests for typing pass.
- Tried to cover +/- cases for each check. Covers:
  - Acquires
  - Type Inference
  - Subtyping

## Test Plan

ran em

### CODE REVIEW

File: language/move-lang/src/typing/expand.rs [hunk 1]
```diff
@@ -257,10 +257,15 @@ fn bind(context: &mut Context, b: &mut T::Bind) {
     use T::Bind_ as B;
     match &mut b.value {
         B::Ignore => (),
-        B::Var(_, st_opt) => {
-            assert!(st_opt.is_some());
-            st_opt.iter_mut().for_each(|st| single_type(context, st))
+        B::Var(v, None) => {
+            let msg = format!(
+                "Unused local '{0}'. Consider removing or prefix with an underscore: '_{0}'",
```
Comment (tzakian) [inline language/move-lang/src/typing/expand.rs]: Is the `{0}` necessary as opposed to `{}`?

Also, what do you think of 'prefixing' as opposed to 'prefix'?

File: language/move-lang/tests/move_check/typing/invalid_type_acquire.exp [hunk 2]
```diff
@@ -0,0 +1,481 @@
...
+
+error: 
+
+    ┌── tests/move_check/typing/invalid_type_acquire.move:18:9 ───
+    │
+ 18 │         T,
+    │         ^ Invalid 'acquires' annotation
+    ·
+ 18 │         T,
+    │         - Expected a nominal resource. Found the type: T
```
Comment (tzakian) [inline language/move-lang/tests/move_check/typing/invalid_type_acquire.exp]: Not needed for this PR, but just a question: What do you think about including the kind of the type in an invalid acquires error setting?

File: language/move-lang/tests/move_check/typing/invalid_type_acquire.exp [hunk 3]
```diff
@@ -0,0 +1,481 @@
...
+
+error: 
+
+    ┌── tests/move_check/typing/invalid_type_acquire.move:33:17 ───
+    │
+ 33 │         destroy(move_from<T>(a));
+    │                 ^^^^^^^^^^^^^^^ Invalid call to move_from.
+    ·
+ 33 │         destroy(move_from<T>(a));
+    │                           - Expected a nominal resource. Found the type: T
```
Comment (tzakian) [inline language/move-lang/tests/move_check/typing/invalid_type_acquire.exp]: Same question here as in the acquires case. It might be helpful (if possible) if the compiler could give the kind of the type at this point.

File: language/move-lang/tests/move_check/typing/missing_acquire.exp [hunk 4]
```diff
@@ -0,0 +1,44 @@
+error: 
+
+    ┌── tests/move_check/typing/missing_acquire.move:8:9 ───
+    │
+ 8 │         r1(a);
+    │         ^^^^^ Invalid call to '0x8675309.M.r1'
+    ·
+ 14 │     r1(a: address) acquires R1 {
+    │                             -- The call acquires '0x8675309.M.R1', but the 'acquires' list for the current function does not contain this type. It must be present to make this call
```
Comment (tzakian) [inline language/move-lang/tests/move_check/typing/missing_acquire.exp]: Nit: The phrase  "must be present to make this call" is pretty ambiguous. Maybe rephrase to something like "must be present in the calling context acquires list" or something along those lines?

File: language/move-lang/src/typing/expand.rs (hunk 1 above)
Comment (tnowacki) [reply language/move-lang/src/typing/expand.rs]: Its used more than once, hence the `{0}`

File: language/move-lang/tests/move_check/typing/invalid_type_acquire.exp (hunk 2 above)
Comment (tnowacki) [reply language/move-lang/tests/move_check/typing/invalid_type_acquire.exp]: A good idea, but another time
